src/internal.c

```diff
@@ -9339,10 +9345,29 @@ static int DoVerifyCallback(WOLFSSL* ssl, int ret, ProcPeerCertArgs* args)
 #endif
 #if defined(OPENSSL_EXTRA)
     /* perform domain name check on the peer certificate */
-    if (args->dCertInit && args->dCert && args->dCert->subjectCN \
-        && ssl->param && ssl->param->hostName[0]) {
+    if (args->dCertInit && args->dCert &&
+            ssl->param && ssl->param->hostName[0]) {
+        /* If altNames names is present, then subject common name is ignored */
+        if (args->dCert->altNames != NULL) {
+            if (CheckAltNames(args->dCert, ssl->param->hostName) == 0 ) {
+                return VERIFY_CERT_ERROR;
```

Comment:
Should we be return ing this early?  What if ALWAYS_VERIFY or VERIFY_CB_ALL is on or if the user wants to override?

Reply:
Agreed. The should check for the status of ret, and if not already an error then store the code to allow the verify CB to review.